ai_guidelines.md

```diff
@@ -0,0 +1,136 @@
+# Guidelines for CommCare HQ
+
+This document outlines recommended practices for working with the CommCare HQ
+codebase. These guidelines apply in the following circumstances:
+
+- Reviewing code or suggesting changes
+- Writing code and making changes
+- Assisting with version control
+
+## General Principles
+
+- Focus on clarity, maintainability, and performance
+- Consider both immediate needs and long-term implications
+
+## Communication
+
+- The AI Assistant should ask clarifying questions whenever details are unclear,
+  or choices are not obvious.
+- When appropriate, the AI Assistant should make recommendations on how the
+  Developer can improve their prompt to get a more effective response.
+
+## Tech Stack
+
+- Backend: Python, Django
+- Python dependency management: uv
+- Testing: pytest
+- Linting & formatting: Flake8, isort, YAPF
+- Frontend: JavaScript, HTMX, Alpine.js, Knockout.js (legacy), Bootstrap
+- JavaScript bundling & dependency management: Webpack, Yarn
+- Databases: PostgreSQL, Elasticsearch, CouchDB (legacy)
+- Asynchronous task queue: Celery
+- Cache & message broker: Redis
+- Stream processing: Kafka
+- Version Control: Git
+
+## Writing and Reviewing Code
+
+### Code Clarity
+
+- Code should be readable and well-structured.
+- Code should follow the conventions used in the rest of the module.
+- When writing or reviewing code, consider any potential bugs or edge cases.
+- Keep functions and methods focused on a single responsibility. Break up
+  longer functions where appropriate.
+
+### Performance
+
+- Consider potential performance bottlenecks.
+- For database operations, queries should be optimized. In rare instances this
+  could require the Developer to test the performance of the query in an
+  environment similar to production. If the AI Assistant is writing code, then
+  notify the Developer when this would be beneficial.
+- For front-end code, be mindful of rendering or loading concerns.
+
+### Maintainability
+
+- Have comments or docstrings that are affected by the change been updated?
```

Comment:
As the other guidelines seem to be declarative, would this be better phrased as "Update comments and docstrings affected by incoming changes"?